ansible/roles/test/tasks/pfc_wd/functional_test/functional_test.yml

```diff
@@ -0,0 +1,344 @@
+
+- set_fact:
+      peer_login: "{{switch_login[hwsku_map[peer_hwsku]]}}"
+
+- name: Set timers
+  set_fact:
+      pfc_wd_detect_time: 200
+      pfc_wd_restore_time: 200
+      pfc_wd_restore_time_large: 30000
+
+- name: Generate config file for drop action.
+  vars:
+      pfc_wd_interface_list: "{{ pfc_wd_test_port }}"
+      pfc_wd_action: drop
+      pfc_wd_detection_time: "{{ pfc_wd_detect_time }}"
+      pfc_wd_restoration_time: "{{ pfc_wd_restore_time_large }}"
+      pfc_wd_op: SET
+  template:
+      src: "{{ pfc_wd_template }}"
+      dest: "{{ tests_location }}/{{testname}}/pfc_wd_drp_action.json"
+  connection: local
+  become: false
+
+- name: Generate config file for timers verification.
+  vars:
+      pfc_wd_interface_list: "{{ pfc_wd_test_port }}"
+      pfc_wd_action: drop
+      pfc_wd_detection_time: "{{ pfc_wd_detect_time }}"
+      pfc_wd_restoration_time: "{{ pfc_wd_restore_time }}"
+      pfc_wd_op: SET
+  template:
+      src: "{{ pfc_wd_template }}"
+      dest: "{{ tests_location }}/{{testname}}/pfc_wd_timer.json"
+  connection: local
+  become: false
+
+- name: Generate config file for forward action.
+  vars:
+      pfc_wd_interface_list: "{{ pfc_wd_test_port }}"
+      pfc_wd_action: forward
+      pfc_wd_detection_time: "{{ pfc_wd_detect_time }}"
+      pfc_wd_restoration_time: "{{ pfc_wd_restore_time_large }}"
+      pfc_wd_op: SET
+  template:
+      src: "{{ pfc_wd_template }}"
+      dest: "{{ tests_location }}/{{testname}}/pfc_wd_fwd_action.json"
+  connection: local
+  become: false
+
+- name: Generate config file for config removal.
+  vars:
+      pfc_wd_interface_list: "{{ pfc_wd_test_port }}"
+      pfc_wd_action: drop
+      pfc_wd_detection_time: "{{ pfc_wd_detect_time }}"
+      pfc_wd_restoration_time: "{{ pfc_wd_restore_time_large }}"
+      pfc_wd_op: DEL
+  template:
+      src: "{{ pfc_wd_template }}"
+      dest: "{{ tests_location }}/{{testname}}/pfc_wd_del_action.json"
+  connection: local
+  become: false
+
+- name: Initialize config test
+  include: roles/test/tasks/init_config_test.yml
+
+- name: copy the test to ptf container
+  copy: src=roles/test/files/ptftests dest=/root
+  delegate_to: "{{ ptf_host }}"
+
+# Verify proper function of drop action
+- block:
+    - name: Apply drop config to {{ pfc_wd_test_port }}.
+      vars:
+        config_file: pfc_wd_drp_action.json
+        errors_expected: false
+        run_cleanup: false
+      include: roles/test/tasks/run_config_test.yml
+
+    - set_fact:
+        test_expect_file: "expect_pfc_wd_detect"
+
+    - name: Initialize loganalyzer
+      include: roles/test/files/tools/loganalyzer/loganalyzer_init.yml
+
+    - name: Generate PFC storm on fanout switch
+      action: apswitch template=pfc_storm.j2
+      args:
+        host: "{{peer_device}}"
+        login: "{{peer_login}}"
+      connection: switch
```

Comment:
pfc_storm.j2 will depends on fanout hwsku. Can we select based on fanout sku?

Reply:
Done